controllers/argocd_metrics_controller.go

```diff
@@ -110,72 +110,135 @@ func (r *ArgoCDMetricsReconciler) Reconcile(ctx context.Context, request reconci
 
 	const clusterMonitoringLabel = "openshift.io/cluster-monitoring"
 	_, exists := namespace.Labels[clusterMonitoringLabel]
-	if !exists {
-		if namespace.Labels == nil {
-			namespace.Labels = make(map[string]string)
+
+	if argocd.Spec.Monitoring.Enabled {
+		if !exists {
+			if namespace.Labels == nil {
+				namespace.Labels = make(map[string]string)
+			}
+			namespace.Labels[clusterMonitoringLabel] = "true"
+			err = r.Client.Update(ctx, &namespace)
+			if err != nil {
+				reqLogger.Error(err, "Error updating namespace",
+					"Namespace", namespace.Name)
+				return reconcile.Result{}, err
+			}
+		} else {
+			namespace.Labels[clusterMonitoringLabel] = "true"
+			err = r.Client.Update(ctx, &namespace)
+			if err != nil {
+				reqLogger.Error(err, "Error updating namespace",
+					"Namespace", namespace.Name)
+				return reconcile.Result{}, err
+			}
+		}
```

Comment:
Is this redundant block intentional?

Reply:
Yes, it the user wants to enable monitoring after disabling it, then in that case we need to re-assign the label and update the namespace. In the first if condition we are creating the label map.

Reply:
Okay. In that case, we can simplify the if  block with something like 

```
        labelVal, exists := namespace.Labels[clusterMonitoringLabel]

	if argocd.Spec.Monitoring.Enabled {
		if !exists || labelVal != "true" {
			if namespace.Labels == nil {
				namespace.Labels = make(map[string]string)
			}
			namespace.Labels[clusterMonitoringLabel] = "true"
			err = r.Client.Update(ctx, &namespace)
			if err != nil {
				reqLogger.Error(err, "Error updating namespace",
					"Namespace", namespace.Name)
				return reconcile.Result{}, err
			}
		} 
```
WDYT?

Reply:
makes sense!